s3/pkg/meta/db/drivers/tidb/bucket.go

```diff
@@ -535,3 +554,78 @@ func (t *TidbClient) ListBucketLifecycle(ctx context.Context) (buckets []*Bucket
 
 	return
 }
+
+func (t *TidbClient) UpdateBucketSSE(ctx context.Context, bucketName string, sseType string, sseserverkey string) error {
+	log.Infof("put bucket[%s] SSE info[%s] into tidb ...\n", bucketName, sseType)
+
+	sql := "update bucket_sseopts set sse=?, sseserverkey=? where bucketname=?"
+	args := []interface{}{sseType, sseserverkey, bucketName}
+
+	_, err := t.Client.Exec(sql, args...)
+	if err != nil {
+		return handleDBError(err)
+	}
+
+	return nil
+}
+
+func (t *TidbClient) CreateBucketSSE(ctx context.Context, bucketName string, sseType string, sseserverkey string) error {
+	log.Infof("create bucket[%s] SSE info[%s] into tidb ...\n", bucketName, sseType)
+
+	sql := "insert into bucket_sseopts(bucketname, sse, sseserverkey) values(?,?,?);"
```

Comment:
@rhsakarpos one question here, why the sse info is saved into a new table bucket_sseopts instead of the bucket table? if the sse info can be saved into bucket table, the bucket_sseopts table is not needed then.
And for Object put operation, we need to save the sse info into object table too, because put object operation can have sse header and it's priority  is higher than the one in bucket.
https://docs.aws.amazon.com/AmazonS3/latest/dev/bucket-encryption.html 
'If your PUT request headers don't include encryption information, Amazon S3 uses the bucket’s default encryption settings to encrypt the objects.

If your PUT request headers include encryption information, Amazon S3 uses the encryption information from the PUT request to encrypt objects before storing them in Amazon S3.'

Reply:
@hopkings2008 yes, this is done to normalize the table design. Rather than storing all info in a single table, with so many columns, we can have a bucket table, and have other tables like bucket_sse and bucket_versioning etc. to store specifics related to these. Makes it easier to extend these features and not affect the basic bucket features

Reply:
Also, for the header comment, you are right. This header is not supported in DAITO. Why ? I need to evaluate ALL cloud backends to see if they support this header (AWS does), then will add to OSDS. Thanks.

Reply:
> @hopkings2008 yes, this is done to normalize the table design. Rather than storing all info in a single table, with so many columns, we can have a bucket table, and have other tables like bucket_sse and bucket_versioning etc. to store specifics related to these. Makes it easier to extend these features and not affect the basic bucket features

@rhsakarpos sse info is specific to that bucket, and the bucket cannot have more than sse settings in the same time, right? so, for this sse case, i think it is not suitable to have one separate table for sse. and also, retrieve that information will need two queries. normalization is used to reduce the level of redundancy, since sse is not redundancy here, so the separate table is not needed. The same as bucket versioning.

Reply:
> Also, for the header comment, you are right. This header is not supported in DAITO. Why ? I need to evaluate ALL cloud backends to see if they support this header (AWS does), then will add to OSDS. Thanks.

sse header in object is defined by s3, users will access multi-cloud through s3 protocol, so multi-cloud needs to be compatible with s3 protocol, api layer must confirm this. if backend doesn't support sse, we can return an error to user.

Reply:
@hopkings2008 , true. However, if the other cloud backend do NOT support this feature of a custom header, then do we still need the access header support ?

Reply:
about the DB table, yes, one bucket has one SSE setting at a time. Duplication apart, there is a separation of concern and low/no impact on existing code, if we keep the schema separate for bucket information, version information, SSE information etc. If any feature adds or removes columns, constraints, primary keys etc. it does not affect the other features

Reply:
one more thing that, if we add one table each time for a new bucket attribute, we will query much time when handle a request, this will introduce a big pressure for the backend db. Also, so many tables will introduce too process logics in a handler, this may introduce problems for maintenance.